linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -3951,6 +3982,24 @@ public void close() {
     }
   }
 
+  private static class CartesianProductJoinEnumerator<TResult, TOuter, TInner>
+      extends CartesianProductEnumerator<Object, TResult> {
+    private final Function2<TOuter, TInner, TResult> resultSelector;
+
+    CartesianProductJoinEnumerator(Function2<TOuter, TInner, TResult> resultSelector,
+                                   Enumerator<Object> outer, Enumerator<Object> inner) {
+      super(ImmutableList.of(outer, inner));
+      this.resultSelector = resultSelector;
+    }
+
+    @Override public TResult current() {
+      final List<Object> list = Arrays.asList(elements.clone());
```

Comment:
What is the purpose of `.clone()` here?

Reply:
Transforming `elements` array into list. Just applying the same approach as `Linq4j#CartesianProductListEnumerator`

Reply:
Note: `list` never escapes this method, so there's no need to copy it.
There's no need to create `list` as well.

Reply:
you're right, I'll change it

Reply:
modified: list no longer generated